app/code/Magento/Theme/Model/Menu/GetCompoundMenu.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Theme\Model\Menu;
+
+use Magento\Theme\Api\GetMenuInterface;
+
+/**
+ * Implementation of GetMenuInterface which merge menu items from several other services.
+ *
+ * @api
+ */
+class GetCompoundMenu implements GetMenuInterface
+{
+    /**
+     * @var GetMenuInterface[]
+     */
+    private $menuProviders;
+
+    /**
+     * GetCompoundMenu constructor.
+     *
+     * @param GetMenuInterface[] $menuProviders
+     */
+    public function __construct(array $menuProviders = [])
+    {
+        $this->menuProviders = array_map(
```

Comment:
I don't think this array_map operation is necessary. As I understand that was introduced to ensure menuProviders type. I'd propose a less strict approach: skip incorrect menu providers inside the "execute" function, dropping a message to system log file.